src/analyze/flavors/custom-element/visit-component-definitions.ts

```diff
@@ -14,7 +14,20 @@ export function visitComponentDefinitions(node: Node, context: VisitComponentDef
 	// customElements.define("my-element", MyElement)
 	if (ts.isCallExpression(node)) {
 		if (ts.isPropertyAccessExpression(node.expression)) {
-			if (node.expression.name != null && ts.isIdentifier(node.expression.name)) {
+			let leftExpression = node.expression.expression;
+			if (
+				ts.isPropertyAccessExpression(leftExpression) &&
+				ts.isIdentifier(leftExpression.expression) &&
+				leftExpression.expression.escapedText === "window"
```

Comment:
usage will be tiny i am sure (currently) but if we must add it, its not difficult.

its really a 'thinking pretty far ahead' edge case IMO and in reality should be avoided since other parsers/tools do the same check as here.